widget/embedded/src/components/Quote/QuoteCostDetails.tsx

```diff
@@ -0,0 +1,236 @@
+import {
+  ChevronDownIcon,
+  ChevronUpIcon,
+  CloseIcon,
+  Divider,
+  IconButton,
+  Modal,
+  QuoteCost,
+  Tooltip,
+  Typography,
+} from '@rango-dev/ui';
+import BigNumber from 'bignumber.js';
+import React, { useRef, useState } from 'react';
+
+import { RANGO_SWAP_BOX_ID } from '../../constants';
+import {
+  GAS_FEE_MAX_DECIMALS,
+  GAS_FEE_MIN_DECIMALS,
+  USD_VALUE_MAX_DECIMALS,
+  USD_VALUE_MIN_DECIMALS,
+} from '../../constants/routing';
+import { useAppStore } from '../../store/AppStore';
+import { getContainer } from '../../utils/common';
+import {
+  formatTooltipNumbers,
+  numberToString,
+  secondsToString,
+  totalArrivalTime,
+} from '../../utils/numbers';
+import {
+  getTotalFeeInUsd,
+  getTotalSwapsFeeInUsd,
+  getUsdSwapsFeeOfStep,
+} from '../../utils/swap';
+
+import {
+  FeeSection,
+  Line,
+  ModalContainer,
+  ModalHeader,
+  Trigger,
+} from './QuoteCostDetails.styles';
+
+const GAS_FEE_MAX = 30;
+
+const NonPayableFee = (props: { fee: BigNumber; label: string }) => {
+  return (
+    !props.fee.isZero() && (
+      <>
+        <FeeSection>
+          <Typography variant="label" size="medium" color="neutral600">
+            {props.label}
+          </Typography>
+          <Typography variant="label" size="medium">
+            $
+            {numberToString(
+              props.fee,
+              USD_VALUE_MIN_DECIMALS,
+              USD_VALUE_MAX_DECIMALS
+            )}
+          </Typography>
+        </FeeSection>
+        <Divider size={8} />
+      </>
+    )
+  );
+};
+
+export function QuoteCostDetails(props: QuoteCostDetailsProps) {
+  const [open, setOpen] = useState<boolean>(false);
+  const [openCollapse, setOpenCollapse] = useState<boolean>(false);
+  const { steps, quote } = props;
+  const tokens = useAppStore().tokens();
+  const totalTime = secondsToString(totalArrivalTime(quote.result?.swaps));
+  const swaps = quote.result?.swaps ?? [];
+  const tooltipContainer = getContainer();
+  const totalNetworkFeeInUsd = getTotalSwapsFeeInUsd(swaps, 'Network Fee');
+  const totalSwappersFeeInUsd = getTotalSwapsFeeInUsd(swaps, 'Swapper Fee');
+  const totalAffiliateFeeInUsd = getTotalSwapsFeeInUsd(swaps, 'Affiliate Fee');
+  const totalOutboundFeeInUsd = getTotalSwapsFeeInUsd(
+    swaps,
+    'Outbound network fee'
+  );
+  const totalRangoFeeInUsd = getTotalSwapsFeeInUsd(swaps, 'Rango Fee');
+
+  const totalFee = getTotalFeeInUsd(swaps, tokens);
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  return (
+    <>
+      <QuoteCost
+        onClickFee={() => setOpen(!open)}
+        fee={numberToString(
+          totalFee,
+          GAS_FEE_MIN_DECIMALS,
+          GAS_FEE_MAX_DECIMALS
+        )}
+        feeWarning={totalFee.gte(new BigNumber(GAS_FEE_MAX))}
+        time={totalTime}
+        steps={steps}
+        tooltipGas={i18n.t('View more info')}
+      />
+
+      <Modal
+        container={document.getElementById(RANGO_SWAP_BOX_ID) || document.body}
```

Comment:
can we get it from the `getContainer` function in `utils/common`?

Reply:
fixed